src/main/src/main/java/org/geoserver/ows/DisabledServiceCheck.java

```diff
@@ -81,6 +112,52 @@ public Service serviceDispatched(Request request, Service service) {
         return service;
     }
 
+    /**
+     * Extracts the context from the request, used to check for layer reference.
+     *
+     * @param request The request.
+     * @return The context.
+     */
+    String context(Request request) {
+        String context = request.getContext();
+        if (context != null && context.contains("gwc/service")) {
+            // Account for context provided to GwcServiceDispatcherCallback
+            if (context.contains("/gwc/service")) {
+                context = context.substring(0, context.indexOf("/gwc/service"));
+            } else {
+                context = context.substring(0, context.indexOf("gwc/service"));
```

Comment:
I guess with subsequent removal of leading and trailing / only this line is needed.

```suggestion
            if (context.contains("gwc/service")) {
                context = context.substring(0, context.indexOf("gwc/service"));
```